students/pkoleyni/Lesson7/html_render.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+
+class Element:
+    tag = 'html'
+    def __init__(self, content=None, **kwargs):
+        self.attr = kwargs
+        print (self.attr)
+        self.content = []
+        if content:
+            self.content.append(content)
+
+    def append(self, new_content):
+        self.content.append(new_content)
+
+    def attr_maker(self):
+        attr_string = " ".join(['{}="{}"'.format(k, v) for k, v in self.attr.items()])
+        if attr_string.strip():
+            open_tag = "<{} {}>".format(self.tag, attr_string.strip())
+        else:
+            open_tag = "<{}>".format(self.tag)
+        close_tag = "</{}>".format(self.tag)
+        return open_tag, close_tag
+
+
+    def render(self, out_file):
+        open_tag, close_tag = self.attr_maker()
+        out_file.write("{}".format(open_tag))
+        for item in self.content:
+            if hasattr(item, "render"):
+                item.render(out_file)
+            else:
+                out_file.write(item)
+                out_file.write("\n")
+        out_file.write("{}".format(close_tag))
+
+
+
+class Html(Element):
+    tag = 'html'
+
+class Body(Element):
+    tag = 'body'
+
+class P(Element):
+    tag = 'p'
+
+class OneLineTag(Element):
+    def render(self, out_file):
+        out_file.write("<{}>".format(self.tag))
+        for item in self.content:
+            out_file.write(item)
+        out_file.write("</{}>".format(self.tag))
```

Comment:
you need to use your `attr_maker` method here - that's why that test fails

when `H(.. )` is called it will `render()` method from `OneLineTag` class and not add attributes.